src/main/java/application/FormplayerAuthFilter.java

```diff
@@ -38,26 +55,78 @@ public void init(FilterConfig filterConfig) throws ServletException {
     @Override
     public void doFilter(ServletRequest req, ServletResponse res,
                          FilterChain chain) throws IOException, ServletException {
-        HttpServletRequest request = (HttpServletRequest) req;
+        FormplayerHttpRequest request = new FormplayerHttpRequest((HttpServletRequest) req);
         if (isAuthorizationRequired(request)) {
-            if (!authorizeRequest(request)) {
+            // These are order dependent
+            if (getSessionId(request) == null) {
                 setResponseUnauthorized((HttpServletResponse) res);
                 return;
             }
+            setToken(request);
+            setUser(request);
+            setDomain(request);
```

Comment:
nice! seems like these three methods could maybe be combined somehow since they share a lot of variables - maybe in the constructor. Let's sync tomorrow